src/plugins/event-plugins/DomEventPlugin.ts

```diff
@@ -121,44 +129,91 @@ export class DomEventPlugin implements Plugin {
+                    elementsList.push(element);
+                }
             });
-        } else if (domEvent.elementId) {
-            document
-                .getElementById(domEvent.elementId)
-                ?.addEventListener(eventType, eventListener);
-        } else if (domEvent.element) {
-            domEvent.element.addEventListener(eventType, eventListener);
+        } else {
+            let element: HTMLElement;
+            if (domEvent.elementId) {
+                if (targetElement && targetElement.id === domEvent.elementId) {
+                    element = targetElement;
+                } else {
+                    element = document.getElementById(domEvent.elementId);
+                }
+            } else if (domEvent.element) {
+                if (targetElement && targetElement === domEvent.element) {
+                    element = targetElement;
+                } else {
+                    element = domEvent.element;
+                }
+            }
+            if (element) {
+                element.addEventListener(eventType, eventListener);
+                elementsList.push(element);
+            }
         }
+
+        elementsList.forEach((element) => {
+            const elementMap: ElementMap = {
+                element,
+                eventListener
+            };
+            if (this.eventListenerMap.has(domEvent)) {
+                this.eventListenerMap.get(domEvent).push(elementMap);
+            } else {
+                this.eventListenerMap.set(domEvent, [elementMap]);
+            }
+        });
     }
 
     private removeEventHandler(domEvent: TargetDomEvent): void {
-        const eventListener:
-            | EventListener
-            | undefined = this.eventListenerMap.get(domEvent);
-
-        if (domEvent.cssLocator && eventListener) {
-            const elementList = document.querySelectorAll(domEvent.cssLocator);
-            elementList.forEach((element: HTMLElement) => {
-                element.removeEventListener(domEvent.event, eventListener);
+        const elementMapList = this.eventListenerMap.get(domEvent);
+        if (elementMapList) {
+            elementMapList.forEach((elementMap) => {
+                elementMap.element.removeEventListener(
+                    domEvent.event,
+                    elementMap.eventListener
+                );
             });
-        } else if (domEvent.elementId && eventListener) {
-            const element = document.getElementById(domEvent.elementId);
-            if (element) {
-                element.removeEventListener(domEvent.event, eventListener);
-            }
-        } else if (domEvent.element && eventListener) {
-            domEvent.element.removeEventListener(domEvent.event, eventListener);
+            this.eventListenerMap.delete(domEvent);
         }
-        this.eventListenerMap.delete(domEvent);
+    }
+
+    private observeDOMMutation() {
+        this.observer = new MutationObserver((mutations) => {
```

Comment:
Yes, `mutationObserver` has option to 'extend monitoring to the entire subtree of nodes rooted at `target`. All of the other properties are then extended to all of the nodes in the subtree instead of applying solely to the target node.' (https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver/observe). We set this option to `true` at the bottom of the method (`this.observer.observe(document, { childList: true, subtree: true });`)